ngraph_bridge/ie_executable.cc

```diff
@@ -91,59 +91,56 @@ IE_Executable::IE_Executable(shared_ptr<Function> func, string device)
   NGRAPH_VLOG(2) << "Loading IE CNN network to device " << m_device;
 
   InferenceEngine::Core ie;
-  // Load model to the plugin (m_device)
+  // Load network to the plugin (m_device) and create an infer request
   InferenceEngine::ExecutableNetwork exe_network =
       ie.LoadNetwork(m_network, m_device);
-
-  // Create infer request
   m_infer_req = exe_network.CreateInferRequest();
 }
 
 bool IE_Executable::call(const vector<shared_ptr<runtime::Tensor>>& outputs,
                          const vector<shared_ptr<runtime::Tensor>>& inputs) {
   InferenceEngine::InputsDataMap input_info = m_network.getInputsInfo();
-  if (input_info.size() != (inputs.size() + m_hoisted_params.size())) {
+  if (input_info.size() != (inputs.size() + m_inputs.size())) {
     THROW_IE_EXCEPTION
         << "Function inputs number differ from number of given inputs";
   }
 
-  //  Prepare input blobs
-  size_t i = 0;
-  for (const auto& it : input_info) {
-    shared_ptr<IETensor> tv;
-    // First check if there were any constants we converted to parameters
-    if (m_hoisted_params.size() > 0) {
-      // We only support one parameter replacement for nullary functions
-      CHECK(m_hoisted_params.size() == 1)
-          << "Multiple input constants were converted to parameters.";
-      CHECK(input_info.size() == 1) << "Expecting one input that was converted "
-                                       "from constant to parameter.";
-      auto input = m_hoisted_params.find(it.first);
-      if (input != m_hoisted_params.end()) {
-        tv = static_pointer_cast<IETensor>((*input).second);
-      } else {
-        THROW_IE_EXCEPTION << "Input value not found for parameter "
-                           << it.first;
-      }
-    } else {
-      tv = static_pointer_cast<IETensor>(inputs[i]);
-    }
-    m_infer_req.SetBlob(it.first, tv->get_blob());
-    i++;
+  auto func = m_network.getFunction();
+  auto parameters = func->get_parameters();
+  for (int i = 0; i < inputs.size(); i++) {
```

Comment:
Why do we need to do this in every tensor execution/call?

Reply:
There will be a different input vector for each call.

Reply:
True, but for a given IE_Executable object, the call param/result names are fixed always.

Reply:
But the map values will change and are provided in this call? So what's the point of a map?